aspnetcore/security/authentication/facebook-logins.md

```diff
@@ -0,0 +1,115 @@
+
+* Tap **My Apps** in the upper right corner:
+
+![image](sociallogins/_static/FBMyApps.png)
+
+* Tap **+ Add a New App** and fill out the form to create a new app ID:
+
+![image](sociallogins/_static/FBNewAppId.png)
+
+* The **Product Setup** page will be displayed, letting you select the features your new app will support. Tap **Get Started** on *Facebook Login*:
+
+![image](sociallogins/_static/FBProductSetup.png)
+
+* You will be presented with the **Client OAuth Settings** page with some defaults already set:
+
+![image](sociallogins/_static/FBOAuthSetup.png)
+
+* Enter your base URI with *signin-facebook* appended into the **Valid OAuth Redirect URIs** field. For example, `https://localhost:44320/signin-facebook`.
+
+* When deploying the site you'll need to register a new public url.
+
+> [!NOTE]
+> You don't need to configure **signin-facebook** as a route in your app. The Facebook middleware will automatically intercept requests at this route and handle them to implement the OAuth flow.
+
+* Make a note of your `App ID` and your `App Secret` before dismissing this page so that you can add both into your ASP.NET Core app later in this tutorial.
+
+* Tap **Save Changes** to complete the new application configuration.
+
+## Storing Facebook AppId and AppSecret
+
+Link sensitive settings like Facebook `App ID` and `App Secret` to your application configuration by using the [Secret Manager tool](../app-secrets.md) instead of storing them in your configuration file directly, as described in the [previous section](sociallogins.md). Execute the following in your project working directory:
+
+* Set the Facebook AppId
+
+  <!-- literal_block {"ids": [], "xml:space": "preserve"} -->
+
+  ````
+  dotnet user-secrets set Authentication:Facebook:AppId <app-Id>
+     ````
+
+* Set the Facebook AppSecret
+
+  <!-- literal_block {"ids": [], "xml:space": "preserve"} -->
+
+  ````
+  dotnet user-secrets set Authentication:Facebook:AppSecret <app-secret>
+     ````
+
+The following code reads the configuration values stored by the [Secret Manager](../app-secrets.md#security-app-secrets):
+
+[!code-csharp[Main](../../common/samples/WebApplication1/Startup.cs?highlight=11&range=20-36)]
+
+## Enable Facebook middleware
+
+> [!NOTE]
+> You will need to use NuGet to install the [Microsoft.AspNetCore.Authentication.Facebook](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Facebook) package if it hasn't already been installed. Alternatively, execute the following in your project directory:
+>
+> `dotnet install Microsoft.AspNetCore.Authentication.Facebook`
+
+Add the Facebook middleware in the `Configure` method in `Startup.cs`:
+
+````csharp
+app.UseFacebookAuthentication(new FacebookOptions()
+{
+    AppId = Configuration["Authentication:Facebook:AppId"],
+    AppSecret = Configuration["Authentication:Facebook:AppSecret"]
+});
+````
+
+## Login with Facebook
+
+Run your application and click Login. You will see an option for Facebook.
```

Comment:
"log in" here is OK because it's in the UI, but style it bold and make it two words, as in the UI.